sklearn/neighbors/graph.py

```diff
@@ -77,7 +79,13 @@ def kneighbors_graph(X, n_neighbors, mode='connectivity', metric='minkowski',
             ).fit(X)
     else:
         _check_params(X, metric, p, metric_params)
-    return X.kneighbors_graph(X._fit_X, n_neighbors, mode=mode)
+    if mode == "connectivity":
+        warnings.warn("The behavior of marking each sample itself as the "
```

Comment:
I guess what I mean is that the API usually gets deprecated: we warn that something that used to be possible will no longer be available once complete. Here, it is not that something will become available, it's that a behaviour will change. I don't think of that as deprecation.